src/machine_helpers.py

```diff
@@ -14,104 +14,45 @@
     TLS_INT_PEM_FILE,
 )
 from charms.mongodb.v0.mongodb import MongoDBConfiguration
-from charms.operator_libs_linux.v1 import systemd
 
 logger = logging.getLogger(__name__)
 
-# systemd gives files in /etc/systemd/system/ precedence over those in /lib/systemd/system/ hence
-# our changed file in /etc will be read while maintaining the original one in /lib.
-MONGOD_SERVICE_UPSTREAM_PATH = "/lib/systemd/system/mongod.service"
-MONGOD_SERVICE_DEFAULT_PATH = "/etc/systemd/system/mongod.service"
+ENV_VAR_PATH = "/etc/environment"
+DB_PROCESS = "/usr/bin/mongod"
+ROOT_USER_GID = 0
+MONGO_USER = "snap_daemon"
 
-# restart options specify that systemd should attempt to restart the service on failure.
-RESTART_OPTIONS = ["Restart=always\n", "RestartSec=20\n"]
-# limits ensure that the process will not continuously retry to restart if it continuously fails to
-# restart.
-RESTARTING_LIMITS = ["StartLimitIntervalSec=500\n", "StartLimitBurst=5\n"]
+MONGODB_COMMON_DIR = "/var/snap/charmed-mongodb/common"
+MONGODB_SNAP_DATA_DIR = "/var/snap/charmed-mongodb/current"
 
-MONGO_USER = "mongodb"
-MONGO_DATA_DIR = "/data/db"
+MONGOD_CONF_DIR = f"{MONGODB_SNAP_DATA_DIR}/etc/mongod"
+MONGOD_CONF_FILE_PATH = f"{MONGOD_CONF_DIR}/mongod.conf"
 
-
-def start_with_auth(path):
-    """Returns true is a mongod service file has the auth configuration."""
-    with open(path, "r") as mongodb_service_file:
-        mongodb_service = mongodb_service_file.readlines()
-
-    for _, line in enumerate(mongodb_service):
-        # ExecStart contains the line with the arguments to start mongod service.
-        if "ExecStart" in line and "--auth" in line:
-            return True
-
-    return False
-
-
-def stop_mongod_service() -> None:
-    """Stop the mongod service if running."""
-    if not systemd.service_running("mongod.service"):
-        return
-
-    logger.debug("stopping mongod.service")
-    try:
-        systemd.service_stop("mongod.service")
-    except systemd.SystemdError as e:
-        logger.error("failed to stop mongod.service, error: %s", str(e))
-        raise
-
-
-def start_mongod_service() -> None:
-    """Starts the mongod service if stopped."""
-    if systemd.service_running("mongod.service"):
-        return
-
-    logger.debug("starting mongod.service")
-    try:
-        systemd.service_start("mongod.service")
-    except systemd.SystemdError as e:
-        logger.error("failed to enable mongod.service, error: %s", str(e))
-        raise
+MONGOD_DATA_DIR = f"{MONGODB_COMMON_DIR}/var/lib/mongod"
+MONGODB_DB_DIR = f"{MONGOD_DATA_DIR}/db"
```

Comment:
```suggestion
MONGODB_DB_DIR = f"{MONGOD_DATA_DIR}"
```

Reply:
We don't want the db directory to be separate? If that's the case, I can remove that entire constant and just use `MONGOD_DATA_DIR`